app/javascript/components/users/SignInComponent.tsx

```diff
@@ -164,84 +165,114 @@ function SignInComponent(props: SignInComponentProps): JSX.Element {
 			}
 				//Props
 			}>{({ isValid }) => {
-				// eslint-disable-next-line react-hooks/rules-of-hooks
 
-				setIsValid(isValid);
+				return (<InnerFormikComponent isSuccessful= {isSuccessful}
+				setIsValid={setIsValid}
+				emailLabel={emailLabel}
+				emailId={emailId}
+				passwordLabel={passwordLabel}
+				passwordId={passwordId}
+				failed={failed}
+				lastSignInAttemptError={lastSignInAttemptError}
+				loading={loading}
+				classes={classes}
+				canSubmit={canSubmit}
+				loginHeaderLabel={loginHeaderLabel}
+				formState={formState}
+				/>);
+			}}
+		</Formik>
+	);
+}
 
+function InnerFormikComponent(props:{isSuccessful:boolean, 
```

Comment:
I split this out into a separate component because there was a warning about using calling `setIsValid` inside the render of Formik. Please move everything that isn't needed in this component down into InnerFormikComponent itself.